src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -433,15 +433,15 @@ public function load(array $configs, ContainerBuilder $container)
         }
 
         if ($this->isConfigEnabled($container, $config['rate_limiter'])) {
-            if (!interface_exists(LimiterInterface::class)) {
+            if (!$container->isAvailable('symfony/rate-limiter', LimiterInterface::class)) {
                 throw new LogicException('Rate limiter support cannot be enabled as the RateLimiter component is not installed. Try running "composer require symfony/rate-limiter".');
             }
 
             $this->registerRateLimiterConfiguration($config['rate_limiter'], $container, $loader);
         }
 
         if ($this->isConfigEnabled($container, $config['web_link'])) {
-            if (!class_exists(HttpHeaderSerializer::class)) {
+            if (!$container->isAvailable('symfony/weblink', HttpHeaderSerializer::class)) {
```

Comment:
IMO, the checks around such exceptions inside a `isConfigEnabled` should keep the old logic. Maybe some projects are indeed enabling some config only in dev environments on purpose (in which case is is OK if it is a dev dependency). The current change forbids enabling the feature at all.

What should be changed instead is the smart default for enabling the config (which is in `Configuration`